salt/modules/seed.py

```diff
@@ -145,7 +145,7 @@ def apply_(path, id_=None, config=None, approve_key=True, install=True,
                  'configuring as {0}'.format(id_))
         minion_config = salt.config.minion_config(cfg_files['config'])
         pki_dir = minion_config['pki_dir']
-        if not os.path.isdir():
+        if not os.path.isdir(os.path.join(mpt, pki_dir)):
```

Comment:
pki_dir starts with a root slash and since path.join pukes on leading edge slashes on anything but the first argument, you need to change..
From
   os.path.join(mpt, pki_dir)
To
   os.path.join(mpt, pki_dir.lstrip('/'))

You have to do the same for the mkdir statement, in addition to giving it a trailing slash.